client/db/bolt/upgrades.go

```diff
@@ -213,6 +216,54 @@ func reloadMatchProofs(tx *bbolt.Tx) error {
 	})
 }
 
+func v4Upgrade(dbtx *bbolt.Tx) error {
+	const oldVersion = 3
+
+	if err := ensureVersion(dbtx, oldVersion); err != nil {
+		return err
+	}
+
+	// Move any executed orders to the new archivedOrdersBucket.
+	return moveArchivedOrders(dbtx)
+}
+
+// moveArchivedOrders searches the ordersBucket for orders that are past booked
+// status, adds those to the new archivedOrdersBucket, and deletes them from the
+// ordersBucket.
+func moveArchivedOrders(tx *bbolt.Tx) error {
+	_, err := tx.CreateBucketIfNotExists(archivedOrdersBucket)
+	if err != nil {
+		return err
+	}
+
+	ob := tx.Bucket(ordersBucket)
+	aob := tx.Bucket(archivedOrdersBucket)
+	return ob.ForEach(func(k, _ []byte) error {
+		oBkt := ob.Bucket(k)
+		if oBkt == nil {
+			return fmt.Errorf("order %x bucket is not a bucket", k)
+		}
+		status := order.OrderStatus(intCoder.Uint16(oBkt.Get(statusKey)))
+		if status == order.OrderStatusUnknown {
+			println(fmt.Sprintf("Encountered order with unknown status: %x", k))
+			return nil
+		}
```

Comment:
Not sure of the best way to handle this.

Reply:
Hmm, I'm thinking Printf an error message and <s>continue to next order</s> dump it into the archived orders bucket even though unknown<booked.  As mentioned in my other comment, I think we should rig all the insert/update functions to refuse to write `OrderStatusUnknown` to a key.